<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="methylclock">
<title>Chronological and gestational DNAm age estimation using different methylation-based clocks • methylclock</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Chronological and gestational DNAm age estimation using different methylation-based clocks">
<meta property="og:description" content="methylclock">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">methylclock</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/methylclock.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/brgelab" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/isglobal-brge/methylclock/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Chronological and gestational DNAm age estimation using different methylation-based clocks</h1>
            <h3 data-toc-skip class="subtitle">Dolors Pelegri and Juan R
Gonzalez</h3>
                        <h4 data-toc-skip class="author"></h4>
<p>Institute
for Global Health (ISGlobal), Barcelona, Spain Bioinformatics Research
Group in Epidemiolgy (BRGE) <a href="http://brge.isglobal.org" class="external-link uri">http://brge.isglobal.org</a></p>
            
            <h4 data-toc-skip class="date">2022-10-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/isglobal-brge/methylclock/blob/HEAD/vignettes/methylclock.Rmd" class="external-link"><code>vignettes/methylclock.Rmd</code></a></small>
      <div class="d-none name"><code>methylclock.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="description-of-implemented-clocks">Description of implemented clocks<a class="anchor" aria-label="anchor" href="#description-of-implemented-clocks"></a>
</h2>
<p>This manual describes how to estimate chronological and gestational
DNA methylation (DNAm) age as well as biological age using different
methylation clocks. The package includes the following estimators:</p>
<div class="section level3">
<h3 id="chronological-dnam-age-in-years">Chronological DNAm age (in years)<a class="anchor" aria-label="anchor" href="#chronological-dnam-age-in-years"></a>
</h3>
<ul>
<li>
<strong>Horvath’s clock</strong>: It uses 353 CpGs described in
<span class="citation">Horvath (2013)</span>. It was trained using 27K
and 450K arrays in samples from different tissues. Other three different
age-related biomarkers are also computed:
<ul>
<li>
<strong>AgeAcDiff</strong> (DNAmAge acceleration difference):
Difference between DNAmAge and chronological age.
<!-- - **IEAA** (Intrinsic Epigenetic Age Acceleration): Residuals obtained  -->
<!-- after regressing DNAmAge and chronological age. -->
<!-- - **EEAA** (Extrinsic Epigenetic Age Acceleration): Residuals obtained  -->
<!-- after regressing DNAmAge and chronological age adjusted by cell counts. -->
<!-- This measure was also known as DNAmAge acceleration residual in the  -->
<!-- first Horvath's paper. -->
</li>
<li>
<strong>IEAA</strong> Residuals obtained after regressing DNAmAge
and chronological age adjusted by cell counts.</li>
<li>
<strong>EEAA</strong> Residuals obtained after regressing DNAmAge
and chronological age. This measure was also known as DNAmAge
acceleration residual in the first Horvath’s paper.</li>
</ul>
</li>
<li>
<strong>Hannum’s clock</strong>: It uses 71 CpGs described in <span class="citation">Hannum et al. (2013)</span>. It was trained using 450K
array in blood samples. Another are-related biomarer is also computed:
<ul>
<li>
<strong>AMAR</strong> (Apparent Methylomic Aging Rate): Measure
proposed in <span class="citation">Hannum et al. (2013)</span> computed
as the ratio between DNAm age and the chronological age.</li>
</ul>
</li>
<li>
<strong>BNN</strong>: It uses Horvath’s CpGs to train a Bayesian
Neural Network (BNN) to predict DNAm age as described in <span class="citation">Alfonso and Gonzalez (2020)</span>.</li>
<li>
<strong>Horvath’s skin+blood clock (skinHorvath)</strong>:
Epigenetic clock for skin and blood cells. It uses 391 CpGs described in
<span class="citation">Horvath et al. (2018)</span>. It was trained
using 450K EPIC arrays in skin and blood sampels.</li>
<li>
<strong>PedBE clock</strong>: Epigenetic clock from buccal
epithelial swabs. It’s intended purpose is buccal samples from
individuals aged 0-20 years old. It uses 84 CpGs described in <span class="citation">McEwen et al. (2019)</span>. The authors gathered 1,721
genome-wide DNAm profiles from 11 different cohorts with individuals
aged 0 to 20 years old.</li>
<li>
<strong>Wu’s clock</strong>: It uses 111 CpGs described in <span class="citation">Wu et al. (2019)</span>. It is designed to predict age
in children. It was trained using 27K and 450K.</li>
<li>
<strong>BLUP clock</strong>: It uses 319607 CpGs described in <span class="citation">Zhang et al. (2019)</span>. It was trained using 450K
and EPIC arrays in blood (13402 samples) and saliva (259 samples). Age
predictors based on training sets with various sample sizes using Best
Linear Unbiased Prediction (BLUP)<br>
</li>
<li>
<strong>EN clock</strong>: It uses 514 CpGs described in <span class="citation">Zhang et al. (2019)</span>. It was trained using 450K
and EPIC arrays in blood (13402 samples) and saliva (259 samples). Age
predictors based on training sets with various sample sizes using
Elastic Net (EN)</li>
</ul>
</div>
<div class="section level3">
<h3 id="gestational-dnam-age-in-weeks">Gestational DNAm age (in weeks)<a class="anchor" aria-label="anchor" href="#gestational-dnam-age-in-weeks"></a>
</h3>
<ul>
<li>
<strong>Knight’s clock</strong>: It uses 148 CpGs described in <span class="citation">Knight et al. (2016)</span>. It was trained using 27K
and 450K arrays in cord blood samples.</li>
<li>
<strong>Bohlin’s clock</strong>: It uses 96 CpGs described in <span class="citation">Bohlin et al. (2016)</span>. It was trained using 450K
array in cord blood samples.</li>
<li>
<strong>Mayne’s clock</strong>: It uses 62 CpGs described in <span class="citation">Mayne et al. (2017)</span>. It was trained using 27K
and 450K.</li>
<li>
<strong>EPIC clock</strong>: EPIC-based predictor of gestational
age. It uses 176 CpGs described in <span class="citation">Haftorn et al.
(2021)</span>. It was trained using EPIC arrays in cord blood
samples.</li>
<li>
<strong>Lee’s clocks</strong>: Three different biological clocks
described in <span class="citation">Lee et al. (2019)</span> are
implemented. It was trained for 450K and EPIC arrays in placenta
samples.</li>
<li>
<strong>RPC clock</strong>: Robust placental clock (RPC). It uses
558 CpG sites.</li>
<li>
<strong>CPC clock</strong>: Control placental clock (CPC). It usses
546 CpG sites.</li>
<li>
<strong>Refined RPC clock</strong>: Useful for uncomplicated term
pregnancies (e.g. gestational age &gt;36 weeks). It uses 396 CpG
sites.</li>
</ul>
<p>The biological DNAm clocks implemented in our package are:</p>
<ul>
<li>
<strong>Levine’s clock</strong> (also know as PhenoAge): It uses 513
CpGs described in <span class="citation">Levine et al. (2018)</span>. It
was trained using 27K, 450K and EPIC arrays in blood samples.</li>
<li>
<strong>Telomere Length’s clock</strong> (TL): It uses 140 CpGs
described in <span class="citation">Lu et al. (2019)</span> It was
trained using 450K and EPIC arrays in blood samples.</li>
</ul>
<p>The main aim of this package is to facilitate the interconnection
with R and Bioconductor’s infrastructure and, hence, avoiding submitting
data to online calculators. Additionally, <code>methylclock</code> also
provides an unified way of computing DNAm age to help downstream
analyses.</p>
</div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>The package depends on some R packages that can be previously
installed into your computer by:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tidyverse"</span>, <span class="st">"impute"</span>, <span class="st">"Rcpp"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then <code>methylclock</code> package is installed into your computer
by executing:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>                        <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"methylclock"</span><span class="op">)</span></span></code></pre></div>
<p>The package is loaded into R as usual:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/isglobal-brge/methylclockData" class="external-link">methylclockData</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/isglobal-brge/methylclock" class="external-link">methylclock</a></span><span class="op">)</span></span></code></pre></div>
<p>These libraries are required to reproduce this document:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biobase" class="external-link">Biobase</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">impute</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.r4photobiology.info/ggpmisc/" class="external-link">ggpmisc</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/seandavi/GEOquery" class="external-link">GEOquery</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dna-methylation-clocks">DNA Methylation clocks<a class="anchor" aria-label="anchor" href="#dna-methylation-clocks"></a>
</h2>
<p>The main function to estimate chronological and biological mDNA age
is called <code>DNAmAge</code> while the gestational DNAm age is
estimated using <code>DNAmGA</code> function. Both functions have
similar input arguments. Next subsections detail some of the important
issues to be consider before computind DNAm clocks.</p>
<div class="section level3">
<h3 id="data-format">Data format<a class="anchor" aria-label="anchor" href="#data-format"></a>
</h3>
<p>The methylation data is given in the argument <code>x</code>. They
can be either beta or M values. The argument <code>toBetas</code> should
be set to TRUE when M values are provided. The <code>x</code> object can
be:</p>
<ul>
<li><p>A <strong>matrix</strong> with CpGs in rows and individuals in
columns having the name of the CpGs in the rownames.</p></li>
<li><p>A <strong>data frame</strong> or a <strong>tibble</strong> with
CpGs in rows and individuals in columns having the name of the CpGs in
the first column (e.g. cg00000292, cg00002426, cg00003994, …) as
required in the Horvath’s DNA Methylation Age Calculator website (<a href="https://dnamage.genetics.ucla.edu/home" class="external-link uri">https://dnamage.genetics.ucla.edu/home</a>).</p></li>
<li><p>A <strong>GenomicRatioSet</strong> object, the default method to
encapsulate methylation data in <code>minfi</code> Bioconductor
package.</p></li>
<li><p>An <strong>ExpressionSet</strong> object as obtained, for
instance, when downloading methylation data from GEO (<a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link uri">https://www.ncbi.nlm.nih.gov/geo/</a>).</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="data-nomalization">Data nomalization<a class="anchor" aria-label="anchor" href="#data-nomalization"></a>
</h3>
<p>In principle, data can be normalized by using any of the existing
standard methods such as QN, ASMN, PBC, SWAN, SQN, BMIQ (see a revision
of those methods in <span class="citation">Wang et al. (2015)</span>).
<code>DNAmAge</code> function includes the BMIQ method proposed by <span class="citation">Teschendorff et al. (2012)</span> using Horvath’s
robust implementation that basically consists of an optimal R code
implementation and optimization procedures. This normalization is
recommended by Horvath since it improves the predictions for his clock.
This normalization procedure is very time-consuming. In order to
overcome these difficulties, we have parallelize this process using
<code>BiocParallel</code> library. This step is not mandatory, so that,
you can use your normalized data and set the argument
<code>normalize</code> equal to FALSE (default).</p>
</div>
<div class="section level3">
<h3 id="missing-individuals-data">Missing individual’s data<a class="anchor" aria-label="anchor" href="#missing-individuals-data"></a>
</h3>
<p>All the implemented methods require complete cases.
<code>DNAmAge</code> function has an imputation method based on KNN
implemented in the function <code>knn.impute</code> from
<code>impute</code> Bioconductor package. This is performed when missing
data is present in the CpGs used in any of the computed clocks. There is
also another option based on a fast imputation method that imputes
missing values by the median of required CpGs as recommended in <span class="citation">Bohlin et al. (2016)</span>. This is recommended when
analyzing 450K arrays since <code>knn.impute</code> for large datasets
may be very time consuming. Fast imputation can be performed by setting
<code>fastImp=TRUE</code> which is not the default value.</p>
</div>
<div class="section level3">
<h3 id="section-missingCpGs">Missing CpGs of DNAm clocks<a class="anchor" aria-label="anchor" href="#section-missingCpGs"></a>
</h3>
<p>By default the package computes the different clocks when there are
more than 80% of the required CpGs of each method. Nothing is required
when having missing CpGs since the main functions will return NA for
those estimators when this criteria is not meet. Let us use a test
dataset (<code>TestDataset</code>) which is available within the package
to illustrate the type of information we are obtaining:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get TestDataset data</span></span>
<span><span class="va">TestDataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/methylclockData/man/get_TestDataset.html" class="external-link">get_TestDataset</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cpgs.missing</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkClocks.html">checkClocks</a></span><span class="op">(</span><span class="va">TestDataset</span><span class="op">)</span></span></code></pre></div>
<pre><code>          clock Cpgs_in_clock missing_CpGs percentage
  1     Horvath           353            2        0.6
  2      Hannum            71           64       90.1
  3      Levine           513            3        0.6
  4 SkinHorvath           391          283       72.4
  5       PedBE            94           91       96.8
  6          Wu           111            2        1.8
  7          TL           140          137       97.9
  8        BLUP        319607       300288       94.0
  9          EN           514          476       92.6</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cpgs.missing.GA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkClocksGA.html">checkClocksGA</a></span><span class="op">(</span><span class="va">TestDataset</span><span class="op">)</span></span></code></pre></div>
<pre><code>     clock Cpgs_in_clock missing_CpGs percentage
  1 Knight           148            0        0.0
  2 Bohlin            87           87      100.0
  3  Mayne            62            0        0.0
  4    Lee          1125         1072       95.3
  5   EPIC           176          170       96.6</code></pre>
<p>The objects <code>cpgs.missing</code> and
<code>cpgs.missing.GA</code> are lists having the missing CpGs of each
clock</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cpgs.missing</span><span class="op">)</span></span></code></pre></div>
<pre><code>  [1] "Horvath"     "Hannum"      "Levine"      "skinHorvath" "PedBE"      
  [6] "Wu"          "TL"          "BLUP"        "EN"</code></pre>
<p>We can see which are those CpGs for a given clock (for example
Hannum) with the function <code>commonClockCpgs</code> :</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/commonClockCpgs.html">commonClockCpgs</a></span><span class="op">(</span><span class="va">cpgs.missing</span>, <span class="st">"Hannum"</span> <span class="op">)</span></span></code></pre></div>
<pre><code>   [1] "cg20822990"      "cg22512670"      "cg25410668"      "cg04400972"     
   [5] "cg16054275"      "cg10501210"      "ch.2.30415474F"  "cg22158769"     
   [9] "cg02085953"      "cg06639320"      "cg22454769"      "cg24079702"     
  [13] "cg23606718"      "cg22016779"      "cg03607117"      "cg07553761"     
  [17] "cg00481951"      "cg25478614"      "cg25428494"      "cg02650266"     
  [21] "cg08234504"      "cg23500537"      "cg20052760"      "cg16867657"     
  [25] "cg06685111"      "cg00486113"      "cg13001142"      "cg20426994"     
  [29] "cg14361627"      "cg08097417"      "cg07955995"      "cg22285878"     
  [33] "cg03473532"      "cg08540945"      "cg07927379"      "cg16419235"     
  [37] "cg07583137"      "cg22796704"      "cg19935065"      "cg23091758"     
  [41] "cg23744638"      "cg04940570"      "cg11067179"      "cg22213242"     
  [45] "cg06419846"      "cg02046143"      "cg00748589"      "cg18473521"     
  [49] "cg01528542"      "ch.13.39564907R" "cg03032497"      "cg04875128"     
  [53] "cg09651136"      "cg03399905"      "cg04416734"      "cg07082267"     
  [57] "cg14692377"      "cg06874016"      "cg21139312"      "cg02867102"     
  [61] "cg19283806"      "cg14556683"      "cg07547549"      "cg08415592"</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/commonClockCpgs.html">commonClockCpgs</a></span><span class="op">(</span><span class="va">cpgs.missing.GA</span>, <span class="st">"Bohlin"</span> <span class="op">)</span></span></code></pre></div>
<pre><code>   [1] "cg00153101" "cg00602416" "cg00711496" "cg01190109" "cg01635555"
   [6] "cg01833485" "cg02324006" "cg02405476" "cg02567958" "cg02642822"
  [11] "cg03108070" "cg03281561" "cg03337084" "cg03507326" "cg03710860"
  [16] "cg03729251" "cg03773820" "cg03963689" "cg04347477" "cg04685228"
  [21] "cg05053327" "cg05544807" "cg05877497" "cg06753281" "cg06897661"
  [26] "cg07106169" "cg07676709" "cg07738730" "cg07749613" "cg07788865"
  [31] "cg07835443" "cg08326019" "cg08620426" "cg08943494" "cg09447786"
  [36] "cg10308785" "cg11124260" "cg11294761" "cg11864574" "cg12880227"
  [41] "cg12999267" "cg13036381" "cg13066703" "cg13433246" "cg13641317"
  [46] "cg13733403" "cg13959344" "cg13982823" "cg14276580" "cg14427590"
  [51] "cg15035133" "cg15131146" "cg15165154" "cg15908709" "cg16187883"
  [56] "cg16348385" "cg17022232" "cg18183624" "cg18217136" "cg18954401"
  [61] "cg19057830" "cg19439123" "cg19875532" "cg20301308" "cg20303561"
  [66] "cg20816447" "cg21081878" "cg21143441" "cg21155834" "cg21221899"
  [71] "cg21707172" "cg21878650" "cg22761205" "cg22796593" "cg22797644"
  [76] "cg23051248" "cg23346945" "cg23403099" "cg23457357" "cg24041556"
  [81] "cg24087613" "cg24366564" "cg25150953" "cg25531857" "cg25639749"
  [86] "cg26077811" "cg26092675"</code></pre>
<p>In Section @ref(section-example) we describe how to change this 80%
threshold.</p>
</div>
<div class="section level3">
<h3 id="cell-counts">Cell counts<a class="anchor" aria-label="anchor" href="#cell-counts"></a>
</h3>
<p>The EEAA method requires to estimate cell counts. We use the package
<code>meffil</code> (<span class="citation">Min et al. (2018)</span>)
that provides some functions to estimate cell counts using predefined
datasets. This is performed by setting <code>cell.count=TRUE</code>
(default value). The reference panel is passed through the argument
<code>cell.count.reference</code>. So far, the following options are
available:</p>
<ul>
<li>
<strong>“blood gse35069 complete”</strong>: methylation profiles
from <span class="citation">Reinius et al. (2012)</span> for purified
blood cell types. It includes CD4T, CD8T, Mono, Bcell, NK, Neu and
Eos.</li>
<li>
<strong>“blood gse35069”</strong>: methylation profiles from <span class="citation">Reinius et al. (2012)</span> for purified blood cell
types. It includes CD4T, CD8T, Mono, Bcell, NK and Gran.</li>
<li>
<strong>“blood gse35069 chen”</strong>: methylation profiles from
<span class="citation">Chen et al. (2017)</span> blood cell types. It
includes CD4T, CD8T, Mono, Bcell, NK, Neu and Eos.</li>
<li>
<strong>“andrews and bakulski cord blood”</strong>. Cord blood
reference from</li>
</ul>
<ol class="example" style="list-style-type: decimal">
<li>It includes Bcell, CD4T, CD8T, Gran, Mono, NK and nRBC.</li>
</ol>
<ul>
<li>
<strong>“cord blood gse68456”</strong> Cord blood methylation
profiles from</li>
</ul>
<ol start="2" class="example" style="list-style-type: decimal">
<li>It includes CD4T, CD8T, Mono, Bcell, NK, Neu, Eos and RBC.</li>
</ol>
<ul>
<li>
<strong>“gervin and lyle cord blood”</strong> Cord blood reference
generated by Kristina Gervin and Robert Lyle, available at
<code>miffil</code> package. It includes CD14, Bcell, CD4T, CD8T, NK,
Gran.</li>
<li>
<strong>“saliva gse48472”</strong>: Reference generated from the
multi-tissue pannel from</li>
</ul>
<ol start="3" class="example" style="list-style-type: decimal">
<li>It includes Buccal, CD4T, CD8T, Mono, Bcell, NK, Gran.</li>
</ol>
<ul>
<li>
<strong>“guintivano dlpfc”</strong>: Reference generated from <span class="citation">Guintivano, Aryee, and Kaminsky (2013)</span>. It
includes dorsolateral prefrontal cortex, NeuN_neg and NeuN_pos.</li>
<li>
<strong>“combined cord blood”</strong>: References generated based
in samples assayed by Bakulski et al, Gervin et al., de Goede et al.,
and Lin et al. It includes umbilical cord blood, Bcell, CD4T, CD8T,
Gran, Mono, NK and nRBC</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="chronological-and-biological-dnam-age-estimation">Chronological and biological DNAm age estimation<a class="anchor" aria-label="anchor" href="#chronological-and-biological-dnam-age-estimation"></a>
</h2>
<p>Next we illustrate how to estimate the chronological DNAm age using
several datasets which aim to cover different data input formats.</p>
<p><strong>IMPORTANT NOTE</strong>: On some systems we can find an error
in the <code><a href="../reference/DNAmAge.html">DNAmAge()</a></code> function when parameter
<code>cell.count = TRUE</code>. This error is related to
<code>preprocessCore</code> package and can be fixed by disabling
multi-threading when installing the preprocessCore package using the
command</p>
<pre><code><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"preprocessCore"</span>, </span>
<span>                     configure.args <span class="op">=</span> <span class="st">"--disable-threading"</span>, </span>
<span>                     force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
<div class="section level3">
<h3 id="section-example">Data in Horvath’s format (e.g. <code>csv</code> with CpGs in
rows)<a class="anchor" aria-label="anchor" href="#section-example"></a>
</h3>
<p>Let us start by reproducing the results proposed in <span class="citation">Horvath (2013)</span>. It uses the format available in
the file ’MethylationDataExample55.csv” from his tutorial (available <a href="https://dnamage.genetics.ucla.edu/home" class="external-link">here</a>). These data are
available at <code>methylclock</code> package. Although these data can
be loaded into R by using standard functions such as
<code>read.csv</code> we hihgly recommend to use functions from
<code>tidiverse</code>, in particular <code>read_csv</code> from
<code>readr</code> package. The main reason is that currently
researchers are analyzing Illumina 450K or EPIC arrays that contains a
huge number of CpGs that can take a long time to be loaded when using
basic importing R function. These functions import <code>csv</code> data
as tibble which is one of the possible formats of <code>DNAmAge</code>
function</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="va">MethylationData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/methylclockData/man/get_MethylationDataExample.html" class="external-link">get_MethylationDataExample</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">MethylationData</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 27,578 × 17 [39m
     ProbeID    GSM946048 GSM946…¹ GSM94…² GSM94…³ GSM94…⁴ GSM94…⁵ GSM94…⁶ GSM94…⁷
      [3m [38;5;246m&lt;chr&gt; [39m [23m           [3m [38;5;246m&lt;dbl&gt; [39m [23m     [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m
   [38;5;250m 1 [39m cg00000292    0.706   0.730   0.705   0.751   0.715   0.634   0.682   0.635  
   [38;5;250m 2 [39m cg00002426    0.272   0.274   0.311   0.279   0.178   0.269   0.330   0.501  
   [38;5;250m 3 [39m cg00003994    0.037 [4m0 [24m  0.014 [4m7 [24m  0.017 [4m1 [24m  0.029 [4m0 [24m  0.016 [4m3 [24m  0.024 [4m3 [24m  0.012 [4m7 [24m  0.020 [4m6 [24m 
   [38;5;250m 4 [39m cg00005847    0.133   0.120   0.121   0.107   0.110   0.129   0.102   0.124  
   [38;5;250m 5 [39m cg00006414    0.030 [4m9 [24m  0.019 [4m2 [24m  0.021 [4m7 [24m  0.013 [4m2 [24m  0.018 [4m1 [24m  0.024 [4m3 [24m  0.019 [4m9 [24m  0.014 [4m3 [24m 
   [38;5;250m 6 [39m cg00007981    0.070 [4m0 [24m  0.071 [4m5 [24m  0.065 [4m5 [24m  0.071 [4m9 [24m  0.091 [4m4 [24m  0.050 [4m8 [24m  0.029 [4m4 [24m  0.056 [4m4 [24m 
   [38;5;250m 7 [39m cg00008493    0.993   0.993   0.993   0.994   0.991   0.994   0.993   0.996  
   [38;5;250m 8 [39m cg00008713    0.021 [4m5 [24m  0.020 [4m2 [24m  0.018 [4m7 [24m  0.016 [4m9 [24m  0.016 [4m2 [24m  0.014 [4m3 [24m  0.017 [4m2 [24m  0.018 [4m9 [24m 
   [38;5;250m 9 [39m cg00009407    0.010 [4m5 [24m  0.005 [4m1 [24m [4m8 [24m 0.004 [4m1 [24m [4m0 [24m 0.006 [4m7 [24m [4m1 [24m 0.007 [4m5 [24m [4m8 [24m 0.005 [4m1 [24m [4m8 [24m 0.005 [4m4 [24m [4m3 [24m 0.006 [4m2 [24m [4m4 [24m
   [38;5;250m10 [39m cg00010193    0.634   0.635   0.621   0.639   0.599   0.591   0.594   0.583  
   [38;5;246m# … with 27,568 more rows, 8 more variables: GSM946064 &lt;dbl&gt;, GSM946065 &lt;dbl&gt;, [39m
   [38;5;246m#   GSM946066 &lt;dbl&gt;, GSM946067 &lt;dbl&gt;, GSM946073 &lt;dbl&gt;, GSM946074 &lt;dbl&gt;, [39m
   [38;5;246m#   GSM946075 &lt;dbl&gt;, GSM946076 &lt;dbl&gt;, and abbreviated variable names [39m
   [38;5;246m#   ¹​GSM946049, ²​GSM946052, ³​GSM946054, ⁴​GSM946055, ⁵​GSM946056, ⁶​GSM946059, [39m
   [38;5;246m#   ⁷​GSM946062 [39m</code></pre>
<p><em>IMPORTANT NOTE</em>: Be sure that the first column contains the
CpG names. Sometimes, your imported data look like this one (it can
happen, for instance, if the <code>csv</code> file was created in R
without indicating <code>row.names=FALSE</code>)</p>
<pre><code>&gt; mydata

# A tibble: 473,999 x 6
    X1 Row.names BIB_15586_1X BIB_33043_1X EDP_5245_1X KAN_584_1X 
    &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;        &lt;dbl&gt;       &lt;dbl&gt;      &lt;dbl&gt;     
1     1 cg000000~       0.635        0.575       0.614      0.631     
2     2 cg000001~       0.954        0.948       0.933      0.950     
3     3 cg000001~       0.889        0.899       0.901      0.892     
4     4 cg000001~       0.115        0.124       0.107      0.123     
5     5 cg000002~       0.850        0.753       0.806      0.815     
6     6 cg000002~       0.676        0.771       0.729      0.665     
7     7 cg000002~       0.871        0.850       0.852      0.863     
8     8 cg000003~       0.238        0.174       0.316      0.206</code></pre>
<p>If so, the first column must be removed before being used as the
input object in <code>DNAmAge</code> funcion. It can be done using
<code>dplyr</code> function</p>
<pre><code>&gt; mydata2 &lt;- select(mydata, -1)

# A tibble: 473,999 x 5
    Row.names BIB_15586_1X BIB_33043_1X EDP_5245_1X KAN_584_1X 
    &lt;chr&gt;            &lt;dbl&gt;        &lt;dbl&gt;       &lt;dbl&gt;      &lt;dbl&gt;     
1    cg000000~       0.635        0.575       0.614      0.631     
2    cg000001~       0.954        0.948       0.933      0.950     
3    cg000001~       0.889        0.899       0.901      0.892     
4    cg000001~       0.115        0.124       0.107      0.123     
5    cg000002~       0.850        0.753       0.806      0.815     
6    cg000002~       0.676        0.771       0.729      0.665     
7    cg000002~       0.871        0.850       0.852      0.863     
8    cg000003~       0.238        0.174       0.316      0.206</code></pre>
<p>In any case, if you use the object <code>mydata</code> that contains
the CpGs in the second column, you will see this error message:</p>
<pre><code>&gt; DNAmAge(mydata)
Error in DNAmAge(mydata) : First column should contain CpG names</code></pre>
<p>Once data is in the proper format, DNAmAge can be estimated by
simply:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">MethylationData</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Warning in predAge(cpgs.imp, coefHannum, intercept = FALSE, min.perc): The number of missing CpGs forHannumclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>   rows : 353 cols : 16</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefSkin, intercept = TRUE, min.perc): The number of missing CpGs forSkinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefPedBE, intercept = TRUE, min.perc): The number of missing CpGs forPedBEclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefTL, intercept = TRUE, min.perc): The number of missing CpGs forTLclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBLUP, intercept = TRUE, min.perc): The number of missing CpGs forBLUPclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEN, intercept = TRUE, min.perc): The number of missing CpGs forENclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 16 × 11 [39m
     id        Horvath Hannum Levine   BNN skinHor…¹ PedBE    Wu TL    BLUP  EN   
      [3m [38;5;246m&lt;chr&gt; [39m [23m        [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m      [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m
   [38;5;250m 1 [39m GSM946048   51.8   [31mNA [39m      - [31m30 [39m [31m. [39m [31m3 [39m 56.4   [31mNA [39m         [31mNA [39m    1.08   [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 2 [39m GSM946049   39.8   [31mNA [39m      - [31m29 [39m [31m. [39m [31m6 [39m 42.1   [31mNA [39m         [31mNA [39m    0.808  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 3 [39m GSM946052   26.4   [31mNA [39m      - [31m33 [39m [31m. [39m [31m3 [39m 25.6   [31mNA [39m         [31mNA [39m    0.772  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 4 [39m GSM946054   34.0   [31mNA [39m      - [31m36 [39m [31m. [39m [31m0 [39m 28.0   [31mNA [39m         [31mNA [39m    0.941  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 5 [39m GSM946055   10.1   [31mNA [39m      - [31m52 [39m [31m. [39m [31m8 [39m 13.4   [31mNA [39m         [31mNA [39m    0.456  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 6 [39m GSM946056   20.4   [31mNA [39m      - [31m42 [39m [31m. [39m [31m2 [39m 16.7   [31mNA [39m         [31mNA [39m    0.621  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 7 [39m GSM946059    6.00  [31mNA [39m      - [31m44 [39m [31m. [39m [31m8 [39m  7.54  [31mNA [39m         [31mNA [39m    0.258  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 8 [39m GSM946062   34.6   [31mNA [39m      - [31m23 [39m [31m. [39m [31m2 [39m 34.6   [31mNA [39m         [31mNA [39m    0.624  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 9 [39m GSM946064    7.91  [31mNA [39m      - [31m49 [39m [31m. [39m [31m8 [39m 12.0   [31mNA [39m         [31mNA [39m    0.237  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m10 [39m GSM946065    4.72  [31mNA [39m      - [31m48 [39m [31m. [39m [31m2 [39m  6.43  [31mNA [39m         [31mNA [39m    0.396  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m11 [39m GSM946066   29.6   [31mNA [39m      - [31m39 [39m [31m. [39m [31m9 [39m 28.5   [31mNA [39m         [31mNA [39m    0.413  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m12 [39m GSM946067    1.38  [31mNA [39m      - [31m48 [39m [31m. [39m [31m3 [39m  3.48  [31mNA [39m         [31mNA [39m    0.122  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m13 [39m GSM946073   56.0   [31mNA [39m      - [31m26 [39m [31m. [39m [31m7 [39m 47.3   [31mNA [39m         [31mNA [39m    0.714  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m14 [39m GSM946074   24.0   [31mNA [39m      - [31m39 [39m [31m. [39m [31m7 [39m 23.3   [31mNA [39m         [31mNA [39m    0.676  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m15 [39m GSM946075    9.38  [31mNA [39m      - [31m45 [39m [31m. [39m [31m4 [39m 11.9   [31mNA [39m         [31mNA [39m    0.251  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m16 [39m GSM946076   38.8   [31mNA [39m      - [31m27 [39m [31m. [39m [31m5 [39m 41.4   [31mNA [39m         [31mNA [39m    0.599  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;246m# … with abbreviated variable name ¹​skinHorvath [39m</code></pre>
<p>As mention in Section @ref(section-missingCpGs) some clocks returns
NA when there are more than 80% of the required CpGs are missing as we
can see when typing</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">missCpGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkClocks.html">checkClocks</a></span><span class="op">(</span><span class="va">MethylationData</span><span class="op">)</span></span></code></pre></div>
<pre><code>          clock Cpgs_in_clock missing_CpGs percentage
  1     Horvath           353            0        0.0
  2      Hannum            71           64       90.1
  3      Levine           513            0        0.0
  4 SkinHorvath           391          282       72.1
  5       PedBE            94           91       96.8
  6          Wu           111            0        0.0
  7          TL           140          137       97.9
  8        BLUP        319607       300192       93.9
  9          EN           514          476       92.6</code></pre>
<p>Here we can observe that 72.1% of the required CpGs for SkinHorvath
clock are missing. We could estimate DNAm age using this clock just
changing the argument <code>min.perc</code> in <code>DNAmAge</code>. For
example, we can indicate that the minimum amount of required CpGs for
computing a given clock should be 25%.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">MethylationData</span>, min.perc <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Warning in predAge(cpgs.imp, coefHannum, intercept = FALSE, min.perc): The number of missing CpGs forHannumclock exceeds 25%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>   rows : 353 cols : 16</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefPedBE, intercept = TRUE, min.perc): The number of missing CpGs forPedBEclock exceeds 25%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefTL, intercept = TRUE, min.perc): The number of missing CpGs forTLclock exceeds 25%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBLUP, intercept = TRUE, min.perc): The number of missing CpGs forBLUPclock exceeds 25%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEN, intercept = TRUE, min.perc): The number of missing CpGs forENclock exceeds 25%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55.2</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 16 × 11 [39m
     id        Horvath Hannum Levine   BNN skinHor…¹ PedBE    Wu TL    BLUP  EN   
      [3m [38;5;246m&lt;chr&gt; [39m [23m        [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m      [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m
   [38;5;250m 1 [39m GSM946048   51.8   [31mNA [39m      - [31m30 [39m [31m. [39m [31m3 [39m 56.4       7.15  [31mNA [39m    1.08   [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 2 [39m GSM946049   39.8   [31mNA [39m      - [31m29 [39m [31m. [39m [31m6 [39m 42.1       7.09  [31mNA [39m    0.808  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 3 [39m GSM946052   26.4   [31mNA [39m      - [31m33 [39m [31m. [39m [31m3 [39m 25.6       5.93  [31mNA [39m    0.772  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 4 [39m GSM946054   34.0   [31mNA [39m      - [31m36 [39m [31m. [39m [31m0 [39m 28.0       6.34  [31mNA [39m    0.941  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 5 [39m GSM946055   10.1   [31mNA [39m      - [31m52 [39m [31m. [39m [31m8 [39m 13.4       5.76  [31mNA [39m    0.456  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 6 [39m GSM946056   20.4   [31mNA [39m      - [31m42 [39m [31m. [39m [31m2 [39m 16.7       5.79  [31mNA [39m    0.621  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 7 [39m GSM946059    6.00  [31mNA [39m      - [31m44 [39m [31m. [39m [31m8 [39m  7.54      5.64  [31mNA [39m    0.258  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 8 [39m GSM946062   34.6   [31mNA [39m      - [31m23 [39m [31m. [39m [31m2 [39m 34.6       5.55  [31mNA [39m    0.624  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 9 [39m GSM946064    7.91  [31mNA [39m      - [31m49 [39m [31m. [39m [31m8 [39m 12.0       5.06  [31mNA [39m    0.237  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m10 [39m GSM946065    4.72  [31mNA [39m      - [31m48 [39m [31m. [39m [31m2 [39m  6.43      5.48  [31mNA [39m    0.396  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m11 [39m GSM946066   29.6   [31mNA [39m      - [31m39 [39m [31m. [39m [31m9 [39m 28.5       6.19  [31mNA [39m    0.413  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m12 [39m GSM946067    1.38  [31mNA [39m      - [31m48 [39m [31m. [39m [31m3 [39m  3.48      4.91  [31mNA [39m    0.122  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m13 [39m GSM946073   56.0   [31mNA [39m      - [31m26 [39m [31m. [39m [31m7 [39m 47.3       7.07  [31mNA [39m    0.714  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m14 [39m GSM946074   24.0   [31mNA [39m      - [31m39 [39m [31m. [39m [31m7 [39m 23.3       6.23  [31mNA [39m    0.676  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m15 [39m GSM946075    9.38  [31mNA [39m      - [31m45 [39m [31m. [39m [31m4 [39m 11.9       5.57  [31mNA [39m    0.251  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m16 [39m GSM946076   38.8   [31mNA [39m      - [31m27 [39m [31m. [39m [31m5 [39m 41.4       6.69  [31mNA [39m    0.599  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;246m# … with abbreviated variable name ¹​skinHorvath [39m</code></pre>
<p>In that case, we see as SkinHorvath clock is estimated (though it can
be observed that the estimation is not very accurate - this is why we
considered at least having 80% of the required CpGs).</p>
<p>By default all available clocks (Hovarth, Hannum, Levine, BNN,
skinHorvath,…) are estimated. One may select a set of clocks by using
the argument <code>clocks</code> as follows:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55.sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">MethylationData</span>, clocks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Horvath"</span>, <span class="st">"BNN"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>   rows : 353 cols : 16</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55.sel</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 16 × 3 [39m
     id        Horvath   BNN
      [3m [38;5;246m&lt;chr&gt; [39m [23m        [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m
   [38;5;250m 1 [39m GSM946048   51.8  56.4 
   [38;5;250m 2 [39m GSM946049   39.8  42.1 
   [38;5;250m 3 [39m GSM946052   26.4  25.6 
   [38;5;250m 4 [39m GSM946054   34.0  28.0 
   [38;5;250m 5 [39m GSM946055   10.1  13.4 
   [38;5;250m 6 [39m GSM946056   20.4  16.7 
   [38;5;250m 7 [39m GSM946059    6.00  7.54
   [38;5;250m 8 [39m GSM946062   34.6  34.6 
   [38;5;250m 9 [39m GSM946064    7.91 12.0 
   [38;5;250m10 [39m GSM946065    4.72  6.43
   [38;5;250m11 [39m GSM946066   29.6  28.5 
   [38;5;250m12 [39m GSM946067    1.38  3.48
   [38;5;250m13 [39m GSM946073   56.0  47.3 
   [38;5;250m14 [39m GSM946074   24.0  23.3 
   [38;5;250m15 [39m GSM946075    9.38 11.9 
   [38;5;250m16 [39m GSM946076   38.8  41.4</code></pre>
</div>
<div class="section level3">
<h3 id="age-acceleration">Age acceleration<a class="anchor" aria-label="anchor" href="#age-acceleration"></a>
</h3>
<p>However, in epidemiological studies one is interested in assessing
whether age acceleration is associated with a given trait or condition.
Three different measures can be computed:</p>
<ul>
<li>
<strong>ageAcc</strong>: Difference between DNAmAge and
chronological age.</li>
<li>
<strong>ageAcc2</strong>: Residuals obtained after regressing
chronological age and DNAmAge (similar to IEAA).</li>
<li>
<strong>ageAcc3</strong>: Residuals obtained after regressing
chronological age and DNAmAge adjusted for cell counts (similar to
EEAA).</li>
</ul>
<p>All this estimates can be obtained for each clock when providing
chronological age through <code>age</code> argument. This information is
normally provided in a different file including different covariates
(metadata or sample annotation data). In this example data are available
at ‘SampleAnnotationExample55.csv’ file that is also available at
<code>methylclock</code> package:</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="va">path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"methylclock"</span><span class="op">)</span></span>
<span><span class="va">covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"SampleAnnotationExample55.csv"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">covariates</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 16 × 14 [39m
     OriginalOr…¹ id    title geo_a…² Tissu…³ Tissue disea…⁴   Age PostM…⁵ Cause…⁶
             [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;chr&gt; [39m [23m  [3m [38;5;246m&lt;chr&gt; [39m [23m  [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m    [3m [38;5;246m&lt;chr&gt; [39m [23m     [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;chr&gt; [39m [23m  
   [38;5;250m 1 [39m            3 GSM9… Auti… GSM946… Fresh … occip…       1    60    26.5 cancer 
   [38;5;250m 2 [39m            4 GSM9… Cont… GSM946… Fresh … occip…       0    39     [31mNA [39m   cardiac
   [38;5;250m 3 [39m            7 GSM9… Auti… GSM946… Fresh … occip…       1    28    43   cancer 
   [38;5;250m 4 [39m            9 GSM9… Auti… GSM946… Fresh … occip…       1    39    14   cardiac
   [38;5;250m 5 [39m           10 GSM9… Auti… GSM946… Fresh … occip…       1     8    22.2 cancer 
   [38;5;250m 6 [39m           11 GSM9… Auti… GSM946… Fresh … occip…       1    22    25   hypoxia
   [38;5;250m 7 [39m           14 GSM9… Cont… GSM946… Fresh … occip…       0     4    17   cardiac
   [38;5;250m 8 [39m           17 GSM9… Cont… GSM946… Fresh … occip…       0    28    13   other  
   [38;5;250m 9 [39m           19 GSM9… Auti… GSM946… Fresh … occip…       1     5    25.5 hypoxia
   [38;5;250m10 [39m           20 GSM9… Auti… GSM946… Fresh … occip…       1     2     4   hypoxia
   [38;5;250m11 [39m           21 GSM9… Auti… GSM946… Fresh … occip…       1    30    16   cardiac
   [38;5;250m12 [39m           22 GSM9… Cont… GSM946… Fresh … occip…       0     1    19   unknown
   [38;5;250m13 [39m           28 GSM9… Cont… GSM946… Fresh … occip…       0    60    24.2 unknown
   [38;5;250m14 [39m           29 GSM9… Cont… GSM946… Fresh … occip…       0    22    21.5 unknown
   [38;5;250m15 [39m           30 GSM9… Cont… GSM946… Fresh … occip…       0     8     5   cardiac
   [38;5;250m16 [39m           31 GSM9… Cont… GSM946… Fresh … occip…       0    30    15   hypoxia
   [38;5;246m# … with 4 more variables: individual &lt;dbl&gt;, Female &lt;dbl&gt;, Caucasian &lt;lgl&gt;, [39m
   [38;5;246m#   FemaleOriginal &lt;lgl&gt;, and abbreviated variable names ¹​OriginalOrder, [39m
   [38;5;246m#   ²​geo_accession, ³​TissueDetailed, ⁴​diseaseStatus, ⁵​PostMortemInterval, [39m
   [38;5;246m#   ⁶​CauseofDeath [39m</code></pre>
<p>In this case, chronological age is available at <code>Age</code>
column:</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age</span> <span class="op">&lt;-</span> <span class="va">covariates</span><span class="op">$</span><span class="va">Age</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">age</span><span class="op">)</span></span></code></pre></div>
<pre><code>  [1] 60 39 28 39  8 22</code></pre>
<p>The different methylation clocks along with their age accelerated
estimates can be simply computed by:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">MethylationData</span>, age<span class="op">=</span><span class="va">age</span>, cell.count<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Warning in predAge(cpgs.imp, coefHannum, intercept = FALSE, min.perc): The number of missing CpGs forHannumclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>   rows : 353 cols : 16</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefSkin, intercept = TRUE, min.perc): The number of missing CpGs forSkinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefPedBE, intercept = TRUE, min.perc): The number of missing CpGs forPedBEclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefTL, intercept = TRUE, min.perc): The number of missing CpGs forTLclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBLUP, intercept = TRUE, min.perc): The number of missing CpGs forBLUPclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEN, intercept = TRUE, min.perc): The number of missing CpGs forENclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.example55</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 16 × 24 [39m
     id      Horvath ageAc…¹ ageAc…² ageAc…³ Hannum Levine ageAc…⁴ ageAc…⁵ ageAc…⁶
      [3m [38;5;246m&lt;chr&gt; [39m [23m      [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m
   [38;5;250m 1 [39m GSM946…   51.8   - [31m8 [39m [31m. [39m [31m22 [39m  - [31m4 [39m [31m. [39m [31m45 [39m    - [31m4 [39m [31m. [39m [31m91 [39m   [31mNA [39m      - [31m30 [39m [31m. [39m [31m3 [39m   - [31m90 [39m [31m. [39m [31m3 [39m  - [31m5 [39m [31m. [39m [31m94 [39m  - [31m6 [39m [31m. [39m [31m19 [39m  
   [38;5;250m 2 [39m GSM946…   39.8    0.754  2.00     1.59   [31mNA [39m      - [31m29 [39m [31m. [39m [31m6 [39m   - [31m68 [39m [31m. [39m [31m6 [39m   3.10   5.38  
   [38;5;250m 3 [39m GSM946…   26.4   - [31m1 [39m [31m. [39m [31m59 [39m  - [31m1 [39m [31m. [39m [31m67 [39m    - [31m1 [39m [31m. [39m [31m86 [39m   [31mNA [39m      - [31m33 [39m [31m. [39m [31m3 [39m   - [31m61 [39m [31m. [39m [31m3 [39m   3.72   2.20  
   [38;5;250m 4 [39m GSM946…   34.0   - [31m5 [39m [31m. [39m [31m00 [39m  - [31m3 [39m [31m. [39m [31m76 [39m    - [31m0 [39m [31m. [39m [31m463 [39m  [31mNA [39m      - [31m36 [39m [31m. [39m [31m0 [39m   - [31m75 [39m [31m. [39m [31m0 [39m  - [31m3 [39m [31m. [39m [31m29 [39m  - [31m0 [39m [31m. [39m [31m339 [39m 
   [38;5;250m 5 [39m GSM946…   10.1    2.06  - [31m0 [39m [31m. [39m [31m428 [39m    2.82   [31mNA [39m      - [31m52 [39m [31m. [39m [31m8 [39m   - [31m60 [39m [31m. [39m [31m8 [39m  - [31m7 [39m [31m. [39m [31m77 [39m   0.964 
   [38;5;250m 6 [39m GSM946…   20.4   - [31m1 [39m [31m. [39m [31m61 [39m  - [31m2 [39m [31m. [39m [31m42 [39m    - [31m2 [39m [31m. [39m [31m88 [39m   [31mNA [39m      - [31m42 [39m [31m. [39m [31m2 [39m   - [31m64 [39m [31m. [39m [31m2 [39m  - [31m2 [39m [31m. [39m [31m76 [39m  - [31m3 [39m [31m. [39m [31m70 [39m  
   [38;5;250m 7 [39m GSM946…    6.00   2.00  - [31m0 [39m [31m. [39m [31m971 [39m   - [31m0 [39m [31m. [39m [31m827 [39m  [31mNA [39m      - [31m44 [39m [31m. [39m [31m8 [39m   - [31m48 [39m [31m. [39m [31m8 [39m   1.81   2.88  
   [38;5;250m 8 [39m GSM946…   34.6    6.65   6.57     5.32   [31mNA [39m      - [31m23 [39m [31m. [39m [31m2 [39m   - [31m51 [39m [31m. [39m [31m2 [39m  13.9    7.97  
   [38;5;250m 9 [39m GSM946…    7.91   2.91   0.058 [4m9 [24m  - [31m2 [39m [31m. [39m [31m61 [39m   [31mNA [39m      - [31m49 [39m [31m. [39m [31m8 [39m   - [31m54 [39m [31m. [39m [31m8 [39m  - [31m3 [39m [31m. [39m [31m61 [39m  - [31m6 [39m [31m. [39m [31m59 [39m  
   [38;5;250m10 [39m GSM946…    4.72   2.72  - [31m0 [39m [31m. [39m [31m489 [39m    1.46   [31mNA [39m      - [31m48 [39m [31m. [39m [31m2 [39m   - [31m50 [39m [31m. [39m [31m2 [39m  - [31m0 [39m [31m. [39m [31m845 [39m  2.53  
   [38;5;250m11 [39m GSM946…   29.6   - [31m0 [39m [31m. [39m [31m427 [39m - [31m0 [39m [31m. [39m [31m268 [39m   - [31m1 [39m [31m. [39m [31m37 [39m   [31mNA [39m      - [31m39 [39m [31m. [39m [31m9 [39m   - [31m69 [39m [31m. [39m [31m9 [39m  - [31m3 [39m [31m. [39m [31m64 [39m  - [31m2 [39m [31m. [39m [31m62 [39m  
   [38;5;250m12 [39m GSM946…    1.38   0.375 - [31m2 [39m [31m. [39m [31m95 [39m    - [31m2 [39m [31m. [39m [31m19 [39m   [31mNA [39m      - [31m48 [39m [31m. [39m [31m3 [39m   - [31m49 [39m [31m. [39m [31m3 [39m  - [31m0 [39m [31m. [39m [31m513 [39m - [31m3 [39m [31m. [39m [31m85 [39m  
   [38;5;250m13 [39m GSM946…   56.0   - [31m4 [39m [31m. [39m [31m0 [39m [31m1 [39m  - [31m0 [39m [31m. [39m [31m242 [39m    1.62   [31mNA [39m      - [31m26 [39m [31m. [39m [31m7 [39m   - [31m86 [39m [31m. [39m [31m7 [39m  - [31m2 [39m [31m. [39m [31m32 [39m  - [31m1 [39m [31m. [39m [31m0 [39m [31m1 [39m  
   [38;5;250m14 [39m GSM946…   24.0    2.03   1.23    - [31m0 [39m [31m. [39m [31m669 [39m  [31mNA [39m      - [31m39 [39m [31m. [39m [31m7 [39m   - [31m61 [39m [31m. [39m [31m7 [39m  - [31m0 [39m [31m. [39m [31m227 [39m  0.021 [4m1 [24m
   [38;5;250m15 [39m GSM946…    9.38   1.38  - [31m1 [39m [31m. [39m [31m11 [39m    - [31m0 [39m [31m. [39m [31m885 [39m  [31mNA [39m      - [31m45 [39m [31m. [39m [31m4 [39m   - [31m53 [39m [31m. [39m [31m4 [39m  - [31m0 [39m [31m. [39m [31m362 [39m - [31m2 [39m [31m. [39m [31m37 [39m  
   [38;5;250m16 [39m GSM946…   38.8    8.76   8.92     5.85   [31mNA [39m      - [31m27 [39m [31m. [39m [31m5 [39m   - [31m57 [39m [31m. [39m [31m5 [39m   8.78   4.73  
   [38;5;246m# … with 14 more variables: BNN &lt;dbl&gt;, ageAcc.BNN &lt;dbl&gt;, ageAcc2.BNN &lt;dbl&gt;, [39m
   [38;5;246m#   ageAcc3.BNN &lt;dbl&gt;, skinHorvath &lt;lgl&gt;, PedBE &lt;lgl&gt;, Wu &lt;dbl&gt;, [39m
   [38;5;246m#   ageAcc.Wu &lt;dbl&gt;, ageAcc2.Wu &lt;dbl&gt;, ageAcc3.Wu &lt;dbl&gt;, TL &lt;lgl&gt;, BLUP &lt;lgl&gt;, [39m
   [38;5;246m#   EN &lt;lgl&gt;, age &lt;dbl&gt;, and abbreviated variable names ¹​ageAcc.Horvath, [39m
   [38;5;246m#   ²​ageAcc2.Horvath, ³​ageAcc3.Horvath, ⁴​ageAcc.Levine, ⁵​ageAcc2.Levine, [39m
   [38;5;246m#   ⁶​ageAcc3.Levine [39m</code></pre>
<p>By default, the argument <code>cell.count</code> is set equal to TRUE
and, hence, can be omitted. This implies that <code>ageAcc3</code> will
be computed for all clocks. In some occassions this can be very time
consuming. In such cases one can simply estimate DNAmAge, accAge and
accAge2 by setting <code>cell.count=FALSE</code>. NOTE: see section 3.5
to see the reference panels available to estimate cell counts.</p>
<p>Then, we can investigate, for instance, whether the accelerated age
is associated with Autism. In that example we will use a non-parametric
test (NOTE: use t-test or linear regression for large sample sizes)</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">autism</span> <span class="op">&lt;-</span> <span class="va">covariates</span><span class="op">$</span><span class="va">diseaseStatus</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/kruskal.test.html" class="external-link">kruskal.test</a></span><span class="op">(</span><span class="va">age.example55</span><span class="op">$</span><span class="va">ageAcc.Horvath</span> <span class="op">~</span> <span class="va">autism</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
    Kruskal-Wallis rank sum test
  
  data:  age.example55$ageAcc.Horvath by autism
  Kruskal-Wallis chi-squared = 1.3346, df = 1, p-value = 0.248</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/kruskal.test.html" class="external-link">kruskal.test</a></span><span class="op">(</span><span class="va">age.example55</span><span class="op">$</span><span class="va">ageAcc2.Horvath</span> <span class="op">~</span> <span class="va">autism</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
    Kruskal-Wallis rank sum test
  
  data:  age.example55$ageAcc2.Horvath by autism
  Kruskal-Wallis chi-squared = 3.1875, df = 1, p-value = 0.0742</code></pre>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/kruskal.test.html" class="external-link">kruskal.test</a></span><span class="op">(</span><span class="va">age.example55</span><span class="op">$</span><span class="va">ageAcc3.Horvath</span> <span class="op">~</span> <span class="va">autism</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
    Kruskal-Wallis rank sum test
  
  data:  age.example55$ageAcc3.Horvath by autism
  Kruskal-Wallis chi-squared = 2.8235, df = 1, p-value = 0.09289</code></pre>
</div>
<div class="section level3">
<h3 id="chronological-age-prediction-using-expressionset-data">Chronological age prediction using <code>ExpressionSet</code>
data<a class="anchor" aria-label="anchor" href="#chronological-age-prediction-using-expressionset-data"></a>
</h3>
<p>One may be interested in assessing association between chronologial
age and DNA methylation age or evaluating how well chronological age is
predicted by DNAmAge. In order to illustrate this analysis we downloaded
data from GEO corresponding to a set of healthy individuals (GEO
accession number GSE58045). Data can be retrieved into R by using
<code>GEOquery</code> package as an <code>ExpressionSet</code> object
that can be the input of our main function.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To avoid connection buffer size </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span><span class="st">"VROOM_CONNECTION_SIZE"</span> <span class="op">=</span> <span class="fl">131072</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download data</span></span>
<span><span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu">GEOquery</span><span class="fu">::</span><span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE58045"</span><span class="op">)</span></span>
<span><span class="va">gse58045</span> <span class="op">&lt;-</span> <span class="va">dd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Restore connection buffer size</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span><span class="st">"VROOM_CONNECTION_SIZE"</span> <span class="op">=</span> <span class="fl">131072</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse58045</span></span></code></pre></div>
<pre><code>  ExpressionSet (storageMode: lockedEnvironment)
  assayData: 27578 features, 172 samples 
    element names: exprs 
  protocolData: none
  phenoData
    sampleNames: GSM1399890 GSM1399891 ... GSM1400061 (172 total)
    varLabels: title geo_accession ... twin:ch1 (43 total)
    varMetadata: labelDescription
  featureData
    featureNames: cg00000292 cg00002426 ... cg27665659 (27578 total)
    fvarLabels: ID Name ... ORF (38 total)
    fvarMetadata: Column Description labelDescription
  experimentData: use 'experimentData(object)'
    pubMedIds: 22532803 
  Annotation: GPL8490</code></pre>
<p>The chronological age is obtained by using <code>pData</code>
function from <code>Biobase</code> package that is able to deal with
<code>ExpressionSet</code> objects:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse58045</span><span class="op">)</span></span>
<span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">`age:ch1`</span><span class="op">)</span></span></code></pre></div>
<p>And the different DNA methylation age estimates are obtained by using
<code>DNAmAge</code> function (NOTE: as there are missing values, the
program automatically runs <code>impute.knn</code> function to get
complete cases):</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.gse58045</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">gse58045</span>, age<span class="op">=</span><span class="va">age</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Imputing missing data of the entire matrix .... 
  Data imputed. Starting DNAm clock estimation ...</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefHannum, intercept = FALSE, min.perc): The number of missing CpGs forHannumclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>   rows : 353 cols : 172</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefSkin, intercept = TRUE, min.perc): The number of missing CpGs forSkinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefPedBE, intercept = TRUE, min.perc): The number of missing CpGs forPedBEclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefTL, intercept = TRUE, min.perc): The number of missing CpGs forTLclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBLUP, intercept = TRUE, min.perc): The number of missing CpGs forBLUPclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEN, intercept = TRUE, min.perc): The number of missing CpGs forENclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.gse58045</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 172 × 24 [39m
     id      Horvath ageAc…¹ ageAc…² ageAc…³ Hannum Levine ageAc…⁴ ageAc…⁵ ageAc…⁶
      [3m [38;5;246m&lt;chr&gt; [39m [23m      [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m
   [38;5;250m 1 [39m GSM139…    65.6   1.07    4.58    5.46   [31mNA [39m       50.7  - [31m13 [39m [31m. [39m [31m8 [39m    2.91    4.28 
   [38;5;250m 2 [39m GSM139…    66.3   0.197   4.06    5.06   [31mNA [39m       51.3  - [31m14 [39m [31m. [39m [31m8 [39m    2.45    5.12 
   [38;5;250m 3 [39m GSM139…    53.9  - [31m5 [39m [31m. [39m [31m31 [39m   - [31m2 [39m [31m. [39m [31m98 [39m   - [31m2 [39m [31m. [39m [31m42 [39m   [31mNA [39m       40.5  - [31m18 [39m [31m. [39m [31m7 [39m   - [31m3 [39m [31m. [39m [31m52 [39m   - [31m2 [39m [31m. [39m [31m54 [39m 
   [38;5;250m 4 [39m GSM139…    40.6  - [31m5 [39m [31m. [39m [31m23 [39m   - [31m5 [39m [31m. [39m [31m89 [39m   - [31m6 [39m [31m. [39m [31m14 [39m   [31mNA [39m       31.3  - [31m14 [39m [31m. [39m [31m5 [39m   - [31m3 [39m [31m. [39m [31m21 [39m   - [31m4 [39m [31m. [39m [31m67 [39m 
   [38;5;250m 5 [39m GSM139…    50.1   0.982   1.06    1.28   [31mNA [39m       41.1   - [31m8 [39m [31m. [39m [31m00 [39m   4.27    3.37 
   [38;5;250m 6 [39m GSM139…    63.7  - [31m0 [39m [31m. [39m [31m895 [39m   2.64    2.92   [31mNA [39m       48.1  - [31m16 [39m [31m. [39m [31m6 [39m    0.223  - [31m1 [39m [31m. [39m [31m25 [39m 
   [38;5;250m 7 [39m GSM139…    44.7  - [31m0 [39m [31m. [39m [31m875 [39m  - [31m1 [39m [31m. [39m [31m59 [39m   - [31m1 [39m [31m. [39m [31m76 [39m   [31mNA [39m       29.2  - [31m16 [39m [31m. [39m [31m4 [39m   - [31m5 [39m [31m. [39m [31m17 [39m   - [31m4 [39m [31m. [39m [31m57 [39m 
   [38;5;250m 8 [39m GSM139…    59.7  - [31m8 [39m [31m. [39m [31m55 [39m   - [31m4 [39m [31m. [39m [31m20 [39m   - [31m3 [39m [31m. [39m [31m48 [39m   [31mNA [39m       41.0  - [31m27 [39m [31m. [39m [31m2 [39m   - [31m9 [39m [31m. [39m [31m37 [39m   - [31m7 [39m [31m. [39m [31m0 [39m [31m6 [39m 
   [38;5;250m 9 [39m GSM139…    48.4  - [31m5 [39m [31m. [39m [31m84 [39m   - [31m4 [39m [31m. [39m [31m63 [39m   - [31m2 [39m [31m. [39m [31m50 [39m   [31mNA [39m       43.8  - [31m10 [39m [31m. [39m [31m4 [39m    3.31    7.18 
   [38;5;250m10 [39m GSM139…    59.3  - [31m3 [39m [31m. [39m [31m93 [39m   - [31m0 [39m [31m. [39m [31m719 [39m  - [31m0 [39m [31m. [39m [31m609 [39m  [31mNA [39m       46.1  - [31m17 [39m [31m. [39m [31m1 [39m   - [31m0 [39m [31m. [39m [31m693 [39m   0.339
   [38;5;246m# … with 162 more rows, 14 more variables: BNN &lt;dbl&gt;, ageAcc.BNN &lt;dbl&gt;, [39m
   [38;5;246m#   ageAcc2.BNN &lt;dbl&gt;, ageAcc3.BNN &lt;dbl&gt;, skinHorvath &lt;lgl&gt;, PedBE &lt;lgl&gt;, [39m
   [38;5;246m#   Wu &lt;dbl&gt;, ageAcc.Wu &lt;dbl&gt;, ageAcc2.Wu &lt;dbl&gt;, ageAcc3.Wu &lt;dbl&gt;, TL &lt;lgl&gt;, [39m
   [38;5;246m#   BLUP &lt;lgl&gt;, EN &lt;lgl&gt;, age &lt;dbl&gt;, and abbreviated variable names [39m
   [38;5;246m#   ¹​ageAcc.Horvath, ²​ageAcc2.Horvath, ³​ageAcc3.Horvath, ⁴​ageAcc.Levine, [39m
   [38;5;246m#   ⁵​ageAcc2.Levine, ⁶​ageAcc3.Levine [39m</code></pre>
<p>Figure <span class="math inline">\(\ref{fig:horvath_age}\)</span>
shows the correlation between DNAmAge obtained from Horvath’s method and
the chronological age, while Figure <span class="math inline">\(\ref{fig:bnn_age}\)</span> depicts the correlation
of a new method based on fitting a Bayesian Neural Network to predict
DNAmAge based on Horvath’s CpGs.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDNAmAge.html">plotDNAmAge</a></span><span class="op">(</span><span class="va">age.gse58045</span><span class="op">$</span><span class="va">Horvath</span>, <span class="va">age</span><span class="op">)</span></span></code></pre></div>
<p><img src="methylclock_files/figure-html/horvat_age-1.png" width="700"></p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDNAmAge.html">plotDNAmAge</a></span><span class="op">(</span><span class="va">age.gse58045</span><span class="op">$</span><span class="va">BNN</span>, <span class="va">age</span>, tit<span class="op">=</span><span class="st">"Bayesian Neural Network"</span><span class="op">)</span></span></code></pre></div>
<p><img src="methylclock_files/figure-html/bnn_age-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="use-of-dnamage-in-association-studies">Use of DNAmAge in association studies<a class="anchor" aria-label="anchor" href="#use-of-dnamage-in-association-studies"></a>
</h3>
<p>Let us illustrate how to use DNAmAge information in association
studies (e.g case/control, smokers/non-smokers,
responders/non-responders, …). GEO number GSE19711 contains
transcriptomic and epigenomic data of a study in lung cancer. Data can
be retrieved into R by</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To avoid connection buffer size </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span><span class="st">"VROOM_CONNECTION_SIZE"</span> <span class="op">=</span> <span class="fl">131072</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download data</span></span>
<span><span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu">GEOquery</span><span class="fu">::</span><span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE19711"</span><span class="op">)</span></span>
<span><span class="va">gse19711</span> <span class="op">&lt;-</span> <span class="va">dd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Restore connection buffer size</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span><span class="st">"VROOM_CONNECTION_SIZE"</span> <span class="op">=</span> <span class="fl">131072</span><span class="op">)</span></span></code></pre></div>
<p>The object <code>gse19711</code>is an <code>ExpressionSet</code> that
can contains CpGs and phenotypic (e.g clinical) information</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gse19711</span></span></code></pre></div>
<pre><code>  ExpressionSet (storageMode: lockedEnvironment)
  assayData: 27578 features, 540 samples 
    element names: exprs 
  protocolData: none
  phenoData
    sampleNames: GSM491937 GSM491938 ... GSM492476 (540 total)
    varLabels: title geo_accession ... stage:ch1 (58 total)
    varMetadata: labelDescription
  featureData
    featureNames: cg00000292 cg00002426 ... cg27665659 (27578 total)
    fvarLabels: ID Name ... ORF (38 total)
    fvarMetadata: Column Description labelDescription
  experimentData: use 'experimentData(object)'
    pubMedIds: 20219944 
  Annotation: GPL8490</code></pre>
<p>Let us imagine we are interested in comparing the accelerated age
between cases and controls. Age and case/control status information can
be obtained by:</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse19711</span><span class="op">)</span></span>
<span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">$</span><span class="va">`ageatrecruitment:ch1`</span><span class="op">)</span></span>
<span><span class="va">disease</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">$</span><span class="va">`sample type:ch1`</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">disease</span><span class="op">)</span></span></code></pre></div>
<pre><code>  disease
     bi-sulphite converted genomic whole blood DNA from Case 
                                                         266 
  bi-sulphite converted genomic whole blood DNA from Control 
                                                         274</code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">disease</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="va">disease</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Control"</span></span>
<span><span class="va">disease</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Case"</span>, <span class="va">disease</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Case"</span></span>
<span><span class="va">disease</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">disease</span>, levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Case"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">disease</span><span class="op">)</span></span></code></pre></div>
<pre><code>  disease
  Control    Case 
      274     266</code></pre>
<p>The DNAmAge estimates of different methods is computed by</p>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.gse19711</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">gse19711</span>, age<span class="op">=</span><span class="va">age</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Imputing missing data of the entire matrix .... 
  Data imputed. Starting DNAm clock estimation ...</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefHannum, intercept = FALSE, min.perc): The number of missing CpGs forHannumclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>   rows : 353 cols : 540</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefSkin, intercept = TRUE, min.perc): The number of missing CpGs forSkinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefPedBE, intercept = TRUE, min.perc): The number of missing CpGs forPedBEclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefTL, intercept = TRUE, min.perc): The number of missing CpGs forTLclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBLUP, intercept = TRUE, min.perc): The number of missing CpGs forBLUPclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEN, intercept = TRUE, min.perc): The number of missing CpGs forENclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<p>We can observe there are missing data. The funcion automatically
impute those using <code>impute.knn</code> function from
<code>impute</code> package since complete cases are required to compute
the different methylation clocks. The estimates are:</p>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">age.gse19711</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 540 × 24 [39m
     id      Horvath ageAc…¹ ageAc…² ageAc…³ Hannum Levine ageAc…⁴ ageAc…⁵ ageAc…⁶
      [3m [38;5;246m&lt;chr&gt; [39m [23m      [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m    [3m [38;5;246m&lt;dbl&gt; [39m [23m
   [38;5;250m 1 [39m GSM491…    62.9   - [31m5 [39m [31m. [39m [31m14 [39m  - [31m0 [39m [31m. [39m [31m351 [39m  - [31m1 [39m [31m. [39m [31m10 [39m   [31mNA [39m       61.1   - [31m6 [39m [31m. [39m [31m90 [39m   7.63     3.65
   [38;5;250m 2 [39m GSM491…    68.8  - [31m12 [39m [31m. [39m [31m2 [39m   - [31m2 [39m [31m. [39m [31m85 [39m   - [31m2 [39m [31m. [39m [31m13 [39m   [31mNA [39m       57.0  - [31m24 [39m [31m. [39m [31m0 [39m   - [31m5 [39m [31m. [39m [31m48 [39m    - [31m1 [39m [31m. [39m [31m54 [39m
   [38;5;250m 3 [39m GSM491…    60.0    3.96   4.54    4.37   [31mNA [39m       43.0  - [31m13 [39m [31m. [39m [31m0 [39m   - [31m2 [39m [31m. [39m [31m13 [39m    - [31m2 [39m [31m. [39m [31m0 [39m [31m3 [39m
   [38;5;250m 4 [39m GSM491…    57.9   - [31m4 [39m [31m. [39m [31m13 [39m  - [31m1 [39m [31m. [39m [31m45 [39m   - [31m1 [39m [31m. [39m [31m38 [39m   [31mNA [39m       40.9  - [31m21 [39m [31m. [39m [31m1 [39m   - [31m8 [39m [31m. [39m [31m40 [39m    - [31m6 [39m [31m. [39m [31m35 [39m
   [38;5;250m 5 [39m GSM491…    59.0  - [31m13 [39m [31m. [39m [31m0 [39m   - [31m6 [39m [31m. [39m [31m79 [39m   - [31m6 [39m [31m. [39m [31m98 [39m   [31mNA [39m       57.0  - [31m15 [39m [31m. [39m [31m0 [39m    0.756   - [31m1 [39m [31m. [39m [31m77 [39m
   [38;5;250m 6 [39m GSM491…    57.0   - [31m4 [39m [31m. [39m [31m00 [39m  - [31m1 [39m [31m. [39m [31m66 [39m   - [31m1 [39m [31m. [39m [31m0 [39m [31m9 [39m   [31mNA [39m       44.7  - [31m16 [39m [31m. [39m [31m3 [39m   - [31m3 [39m [31m. [39m [31m92 [39m    - [31m1 [39m [31m. [39m [31m35 [39m
   [38;5;250m 7 [39m GSM491…    61.9   - [31m3 [39m [31m. [39m [31m0 [39m [31m8 [39m   0.657   0.183  [31mNA [39m       47.9  - [31m17 [39m [31m. [39m [31m1 [39m   - [31m3 [39m [31m. [39m [31m47 [39m    - [31m4 [39m [31m. [39m [31m21 [39m
   [38;5;250m 8 [39m GSM491…    59.1  - [31m11 [39m [31m. [39m [31m9 [39m   - [31m6 [39m [31m. [39m [31m0 [39m [31m7 [39m   - [31m5 [39m [31m. [39m [31m53 [39m   [31mNA [39m       50.0  - [31m21 [39m [31m. [39m [31m0 [39m   - [31m5 [39m [31m. [39m [31m60 [39m    - [31m3 [39m [31m. [39m [31m0 [39m [31m6 [39m
   [38;5;250m 9 [39m GSM491…    60.7  - [31m16 [39m [31m. [39m [31m3 [39m   - [31m8 [39m [31m. [39m [31m33 [39m   - [31m9 [39m [31m. [39m [31m33 [39m   [31mNA [39m       47.7  - [31m29 [39m [31m. [39m [31m3 [39m  - [31m12 [39m [31m. [39m [31m0 [39m    - [31m15 [39m [31m. [39m [31m7 [39m 
   [38;5;250m10 [39m GSM491…    51.1   - [31m7 [39m [31m. [39m [31m93 [39m  - [31m6 [39m [31m. [39m [31m30 [39m   - [31m6 [39m [31m. [39m [31m33 [39m   [31mNA [39m       52.5   - [31m6 [39m [31m. [39m [31m54 [39m   5.26     1.30
   [38;5;246m# … with 530 more rows, 14 more variables: BNN &lt;dbl&gt;, ageAcc.BNN &lt;dbl&gt;, [39m
   [38;5;246m#   ageAcc2.BNN &lt;dbl&gt;, ageAcc3.BNN &lt;dbl&gt;, skinHorvath &lt;lgl&gt;, PedBE &lt;lgl&gt;, [39m
   [38;5;246m#   Wu &lt;dbl&gt;, ageAcc.Wu &lt;dbl&gt;, ageAcc2.Wu &lt;dbl&gt;, ageAcc3.Wu &lt;dbl&gt;, TL &lt;lgl&gt;, [39m
   [38;5;246m#   BLUP &lt;lgl&gt;, EN &lt;lgl&gt;, age &lt;dbl&gt;, and abbreviated variable names [39m
   [38;5;246m#   ¹​ageAcc.Horvath, ²​ageAcc2.Horvath, ³​ageAcc3.Horvath, ⁴​ageAcc.Levine, [39m
   [38;5;246m#   ⁵​ageAcc2.Levine, ⁶​ageAcc3.Levine [39m</code></pre>
<p>The association between disease status and DNAmAge estimated using
Horvath’s method can be computed by</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod.horvath1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc.Horvath</span> , </span>
<span>                    data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                    family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.horvath1</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc.Horvath, family = "binomial", 
      data = age.gse19711)
  
  Deviance Residuals: 
     Min      1Q  Median      3Q     Max  
  -1.358  -1.160  -1.030   1.184   1.771  
  
  Coefficients:
                 Estimate Std. Error z value Pr(&gt;|z|)  
  (Intercept)    -0.10995    0.09771  -1.125   0.2605  
  ageAcc.Horvath -0.02023    0.01154  -1.753   0.0795 .
  ---
  Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 745.25  on 538  degrees of freedom
  AIC: 749.25
  
  Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod.skinHorvath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc2.Horvath</span> , </span>
<span>                       data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                       family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.skinHorvath</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc2.Horvath, family = "binomial", 
      data = age.gse19711)
  
  Deviance Residuals: 
     Min      1Q  Median      3Q     Max  
  -1.279  -1.163  -1.082   1.189   1.589  
  
  Coefficients:
                  Estimate Std. Error z value Pr(&gt;|z|)
  (Intercept)     -0.02970    0.08617  -0.345    0.730
  ageAcc2.Horvath -0.01315    0.01209  -1.087    0.277
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 747.27  on 538  degrees of freedom
  AIC: 751.27
  
  Number of Fisher Scoring iterations: 3</code></pre>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod.horvath3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc3.Horvath</span> , </span>
<span>                    data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                    family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.horvath3</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc3.Horvath, family = "binomial", 
      data = age.gse19711)
  
  Deviance Residuals: 
     Min      1Q  Median      3Q     Max  
  -1.338  -1.163  -1.046   1.185   1.771  
  
  Coefficients:
                  Estimate Std. Error z value Pr(&gt;|z|)
  (Intercept)     -0.02993    0.08626  -0.347    0.729
  ageAcc3.Horvath -0.01927    0.01283  -1.502    0.133
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 746.13  on 538  degrees of freedom
  AIC: 750.13
  
  Number of Fisher Scoring iterations: 4</code></pre>
<p>We do not observe statistical significant association between age
acceleration estimated using Horvath method and the risk of developing
lung cancer. It is worth to notice that Horvath’s clock was created to
predict chronological age and the impact of age acceleration of this
clock on disease may be limited. On the other hand, Levine’s clock aimed
to distinguish risk between same-aged individuals. Let us evaluate
whether this age acceleration usin Levine’s clock is associated with
lung cancer</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod.levine1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc.Levine</span> , data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                    family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.levine1</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc.Levine, family = "binomial", data = age.gse19711)
  
  Deviance Residuals: 
     Min      1Q  Median      3Q     Max  
  -1.592  -1.149  -0.939   1.174   1.733  
  
  Coefficients:
                Estimate Std. Error z value Pr(&gt;|z|)   
  (Intercept)    0.40956    0.17894   2.289  0.02209 * 
  ageAcc.Levine  0.03178    0.01133   2.806  0.00502 **
  ---
  Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 740.17  on 538  degrees of freedom
  AIC: 744.17
  
  Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod.levine2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc2.Levine</span> , data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                    family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.levine2</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc2.Levine, family = "binomial", 
      data = age.gse19711)
  
  Deviance Residuals: 
      Min       1Q   Median       3Q      Max  
  -1.7053  -1.1328  -0.8614   1.1529   1.8015  
  
  Coefficients:
                 Estimate Std. Error z value Pr(&gt;|z|)    
  (Intercept)    -0.02925    0.08718  -0.336 0.737225    
  ageAcc2.Levine  0.04430    0.01234   3.589 0.000332 ***
  ---
  Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 734.49  on 538  degrees of freedom
  AIC: 738.49
  
  Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod.levine3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc3.Levine</span> , data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                    family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.levine3</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc3.Levine, family = "binomial", 
      data = age.gse19711)
  
  Deviance Residuals: 
     Min      1Q  Median      3Q     Max  
  -1.354  -1.161  -1.057   1.187   1.408  
  
  Coefficients:
                 Estimate Std. Error z value Pr(&gt;|z|)
  (Intercept)    -0.02962    0.08622  -0.344    0.731
  ageAcc3.Levine  0.01679    0.01244   1.350    0.177
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 746.62  on 538  degrees of freedom
  AIC: 750.62
  
  Number of Fisher Scoring iterations: 3</code></pre>
<p>Here we observe as the risk of developing lung cancer increases 3.23
percent per each unit in the age accelerated variable
(<code>ageAcc</code>). Similar conclusion is obtained when using
<code>ageAcc2</code> and <code>ageAcc3</code> variables.</p>
<p>In some occasions cell composition should be used to assess
association. This information is calculated in <code>DNAmAge</code>
function and it can be incorporated in the model by:</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">age.gse19711</span>, <span class="st">"cell_proportion"</span><span class="op">)</span></span>
<span><span class="va">mod.cell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">ageAcc.Levine</span> <span class="op">+</span> <span class="va">cell</span>, data<span class="op">=</span><span class="va">age.gse19711</span>,</span>
<span>                    family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mod.cell</span><span class="op">)</span></span></code></pre></div>
<pre><code>  
  Call:
  glm(formula = disease ~ ageAcc.Levine + cell, family = "binomial", 
      data = age.gse19711)
  
  Deviance Residuals: 
      Min       1Q   Median       3Q      Max  
  -1.9605  -1.0832  -0.6241   1.0742   2.3395  
  
  Coefficients:
                 Estimate Std. Error z value Pr(&gt;|z|)    
  (Intercept)   -9.768206   4.380382  -2.230 0.025748 *  
  ageAcc.Levine  0.003959   0.012208   0.324 0.745746    
  cellCD4T      -3.339693   3.833531  -0.871 0.383656    
  cellMono      10.165096   4.594096   2.213 0.026922 *  
  cellNeu       16.319534   4.584745   3.560 0.000372 ***
  cellNK        -0.882134   4.296498  -0.205 0.837326    
  ---
  Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
  
  (Dispersion parameter for binomial family taken to be 1)
  
      Null deviance: 748.48  on 539  degrees of freedom
  Residual deviance: 686.56  on 534  degrees of freedom
  AIC: 698.56
  
  Number of Fisher Scoring iterations: 4</code></pre>
<p>Here we observe as the positive association disapears after adjusting
for cell counts.</p>
</div>
<div class="section level3">
<h3 id="use-of-dnam-age-in-children">Use of DNAm age in children<a class="anchor" aria-label="anchor" href="#use-of-dnam-age-in-children"></a>
</h3>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu">GEOquery</span><span class="fu">::</span><span class="fu"><a href="http://seandavi.github.io/GEOquery/reference/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE109446"</span><span class="op">)</span></span>
<span><span class="va">gse109446</span> <span class="op">&lt;-</span> <span class="va">dd</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">controls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse109446</span><span class="op">)</span><span class="op">$</span><span class="va">`diagnosis:ch1`</span><span class="op">==</span><span class="st">"control"</span></span>
<span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="va">gse109446</span><span class="op">[</span>,<span class="va">controls</span><span class="op">]</span></span>
<span><span class="va">age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span><span class="op">$</span><span class="va">`age:ch1`</span><span class="op">)</span></span>
<span><span class="va">age.gse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmAge.html">DNAmAge</a></span><span class="op">(</span><span class="va">gse</span>, age<span class="op">=</span><span class="va">age</span><span class="op">)</span></span></code></pre></div>
<pre><code>   rows : 353 cols : 29</code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCorClocks.html">plotCorClocks</a></span><span class="op">(</span><span class="va">age.gse</span><span class="op">)</span></span></code></pre></div>
<p><img src="methylclock_files/figure-html/plotClocks-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="gestational-dnam-age-estimation">Gestational DNAm age estimation<a class="anchor" aria-label="anchor" href="#gestational-dnam-age-estimation"></a>
</h2>
<div class="section level3">
<h3 id="model-predicion">Model predicion<a class="anchor" aria-label="anchor" href="#model-predicion"></a>
</h3>
<p>Let us start by reproducing the example provided in <span class="citation">Knight et al. (2016)</span> as a test data set (file
‘TestDataset.csv’). It consists on 3 individuals whose methylation data
are available as supplementary data of their paper. The data is also
available at <code>methylclock</code> package as a data frame.</p>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TestDataset</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<pre><code>       CpGName    Sample1    Sample2    Sample3
  1 cg00000292 0.72546496 0.72350947 0.69023377
  2 cg00002426 0.85091763 0.80077888 0.80385777
  3 cg00003994 0.05125853 0.05943935 0.05559333
  4 cg00005847 0.08775420 0.11722333 0.10845113
  5 cg00006414 0.03982478 0.06146891 0.03491992</code></pre>
<p>The Gestational Age (in months) is simply computed by</p>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ga.test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmGA.html">DNAmGA</a></span><span class="op">(</span><span class="va">TestDataset</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Warning in DNAmGA(TestDataset): CpGs in all Gestational Age clocks are not present in your
          data. Try 'checkClocksGA' function to find the missing CpGs of
                  each method.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBohlin, intercept = TRUE, min.perc): The number of missing CpGs forBohlinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEPIC, intercept = TRUE, min.perc): The number of missing CpGs forEPICclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in DNAmGA(TestDataset): The number of missing CpGs for Lee clocks exceeds 80%.
                  
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ga.test</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 3 × 6 [39m
    id      Knight Bohlin Mayne EPIC  Lee  
     [3m [38;5;246m&lt;chr&gt; [39m [23m     [3m [38;5;246m&lt;dbl&gt; [39m [23m   [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m
   [38;5;250m1 [39m Sample1   38.2      [31mNA [39m  35.8  [31mNA [39m     [31mNA [39m   
   [38;5;250m2 [39m Sample2   38.8      [31mNA [39m  36.5  [31mNA [39m     [31mNA [39m   
   [38;5;250m3 [39m Sample3   40.0      [31mNA [39m  36.6  [31mNA [39m     [31mNA [39m</code></pre>
<p>like in DNAmAge we can use the parameter <code>min.perc</code> to set
the minimum missing percentage.</p>
<p>The results are the same as those described in the additional file 7
of <span class="citation">Knight et al. (2016)</span> (link [here] (<a href="https://static-content.springer.com/esm/art%3A10.1186%2Fs13059-016-1068-z/MediaObjects/13059_2016_1068_MOESM7_ESM.docx" class="external-link uri">https://static-content.springer.com/esm/art%3A10.1186%2Fs13059-016-1068-z/MediaObjects/13059_2016_1068_MOESM7_ESM.docx</a>))</p>
<p>Let us continue by illustrating how to compute GA of real examples.
The PROGRESS cohort data is available in the additional file 8 of <span class="citation">Knight et al. (2016)</span>. It is available at
<code>methylclock</code> as a <code>tibble</code>:</p>
<div class="sourceCode" id="cb128"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">progress_data</span><span class="op">)</span></span></code></pre></div>
<p>This file also contains different variables that are available in
this <code>tibble</code>.</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">progress_vars</span><span class="op">)</span></span></code></pre></div>
<p>The Clinical Variables including clinical assesment of gestational
age (EGA) are available at this <code>tibble</code>.</p>
<p>The Gestational Age (in months) is simply computed by</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ga.progress</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmGA.html">DNAmGA</a></span><span class="op">(</span><span class="va">progress_data</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Warning in DNAmGA(progress_data): CpGs in all Gestational Age clocks are not present in your
          data. Try 'checkClocksGA' function to find the missing CpGs of
                  each method.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBohlin, intercept = TRUE, min.perc): The number of missing CpGs forBohlinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefMayneGA, intercept = TRUE, min.perc): The number of missing CpGs forMayneclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEPIC, intercept = TRUE, min.perc): The number of missing CpGs forEPICclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in DNAmGA(progress_data): The number of missing CpGs for Lee clocks exceeds 80%.
                  
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ga.progress</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 150 × 6 [39m
     id    Knight Bohlin Mayne EPIC  Lee  
      [3m [38;5;246m&lt;chr&gt; [39m [23m   [3m [38;5;246m&lt;dbl&gt; [39m [23m   [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m
   [38;5;250m 1 [39m 784     38.8      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 2 [39m 1052    37.2      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 3 [39m 1048    40.3      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 4 [39m 1017    39.2      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 5 [39m 956     38.9      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 6 [39m 1038    39.2      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 7 [39m 989     37.2      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 8 [39m 946     35.4      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m 9 [39m 941     33.5      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;250m10 [39m 1024    37.4      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m   
   [38;5;246m# … with 140 more rows [39m</code></pre>
<p>We can compare these results with the clinical GA available in the
variable EGA</p>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotDNAmAge.html">plotDNAmAge</a></span><span class="op">(</span><span class="va">ga.progress</span><span class="op">$</span><span class="va">Knight</span>, <span class="va">progress_vars</span><span class="op">$</span><span class="va">EGA</span>, </span>
<span>            tit<span class="op">=</span><span class="st">"GA Knight's method"</span>, </span>
<span>            clock<span class="op">=</span><span class="st">"GA"</span><span class="op">)</span></span></code></pre></div>
<p><img src="methylclock_files/figure-html/plot_progress-1.png" width="700"></p>
<p>Figure 3b (only for PROGRESS dataset) in <span class="citation">Knight et al. (2016)</span> representing the
correlation between GA acceleration and birthweight can be reproduced
by</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">progress_vars</span><span class="op">$</span><span class="va">acc</span> <span class="op">&lt;-</span> <span class="va">ga.progress</span><span class="op">$</span><span class="va">Knight</span> <span class="op">-</span> <span class="va">progress_vars</span><span class="op">$</span><span class="va">EGA</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">progress_vars</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">acc</span>, y <span class="op">=</span> <span class="va">birthweight</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">geom_smooth</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"lm"</span>, se<span class="op">=</span><span class="cn">FALSE</span>, color<span class="op">=</span><span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"GA acceleration"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Birthweight (kgs.)"</span><span class="op">)</span> </span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="methylclock_files/figure-html/plotAcc-1.png" width="700"></p>
<p>Finally, we can also estimate the “accelerated gestational age” using
two of the the three different estimates previously described
(<code>accAge</code>, <code>accAge2</code>) by provinding information of
gestational age through <code>age</code> argument. Notice that in that
case <code>accAge3</code> cannot be estimates since we do not have all
the CpGs required by the default reference panel to estimate cell counts
for gestational age which is “andrews and bakulski cord blood”.</p>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">accga.progress</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DNAmGA.html">DNAmGA</a></span><span class="op">(</span><span class="va">progress_data</span>, </span>
<span>                        age <span class="op">=</span> <span class="va">progress_vars</span><span class="op">$</span><span class="va">EGA</span>, </span>
<span>                        cell.count<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Warning in DNAmGA(progress_data, age = progress_vars$EGA, cell.count = FALSE): CpGs in all Gestational Age clocks are not present in your
          data. Try 'checkClocksGA' function to find the missing CpGs of
                  each method.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefBohlin, intercept = TRUE, min.perc): The number of missing CpGs forBohlinclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefMayneGA, intercept = TRUE, min.perc): The number of missing CpGs forMayneclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in predAge(cpgs.imp, coefEPIC, intercept = TRUE, min.perc): The number of missing CpGs forEPICclock exceeds 80%.
    ---&gt; This DNAm clock will be NA.</code></pre>
<pre><code>  Warning in DNAmGA(progress_data, age = progress_vars$EGA, cell.count = FALSE): The number of missing CpGs for Lee clocks exceeds 80%.
                  
    ---&gt; This DNAm clock will be NA.</code></pre>
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">accga.progress</span></span></code></pre></div>
<pre><code>   [38;5;246m# A tibble: 150 × 9 [39m
     id    Knight ageAcc.Knight ageAcc2.Knight Bohlin Mayne EPIC  Lee     age
      [3m [38;5;246m&lt;chr&gt; [39m [23m   [3m [38;5;246m&lt;dbl&gt; [39m [23m          [3m [38;5;246m&lt;dbl&gt; [39m [23m           [3m [38;5;246m&lt;dbl&gt; [39m [23m   [3m [38;5;246m&lt;dbl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;lgl&gt; [39m [23m  [3m [38;5;246m&lt;dbl&gt; [39m [23m
   [38;5;250m 1 [39m 784     38.8         0.792          1.27       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38  
   [38;5;250m 2 [39m 1052    37.2        - [31m1 [39m [31m. [39m [31m0 [39m [31m5 [39m          - [31m0 [39m [31m. [39m [31m488 [39m      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38.3
   [38;5;250m 3 [39m 1048    40.3         2.29           2.77       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38  
   [38;5;250m 4 [39m 1017    39.2         0.643          1.28       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38.6
   [38;5;250m 5 [39m 956     38.9         1.75           1.99       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     37.1
   [38;5;250m 6 [39m 1038    39.2         1.09           1.61       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38.1
   [38;5;250m 7 [39m 989     37.2        - [31m0 [39m [31m. [39m [31m774 [39m         - [31m0 [39m [31m. [39m [31m292 [39m      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38  
   [38;5;250m 8 [39m 946     35.4        - [31m2 [39m [31m. [39m [31m36 [39m          - [31m1 [39m [31m. [39m [31m96 [39m       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     37.7
   [38;5;250m 9 [39m 941     33.5        - [31m3 [39m [31m. [39m [31m18 [39m          - [31m3 [39m [31m. [39m [31m0 [39m [31m6 [39m       [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     36.7
   [38;5;250m10 [39m 1024    37.4        - [31m1 [39m [31m. [39m [31m12 [39m          - [31m0 [39m [31m. [39m [31m486 [39m      [31mNA [39m  [31mNA [39m     [31mNA [39m     [31mNA [39m     38.6
   [38;5;246m# … with 140 more rows [39m</code></pre>
<p>One can also check which clocks can be estimated given the CpGs
available in the methylation data by</p>
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/checkClocksGA.html">checkClocksGA</a></span><span class="op">(</span><span class="va">progress_data</span><span class="op">)</span></span></code></pre></div>
<pre><code>     clock Cpgs_in_clock missing_CpGs percentage
  1 Knight           148            0        0.0
  2 Bohlin            94           94      100.0
  3  Mayne            62           61       98.4
  4    Lee          1125         1125      100.0
  5   EPIC           176          176      100.0</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="correlation-among-dnam-clocks">Correlation among DNAm clocks<a class="anchor" aria-label="anchor" href="#correlation-among-dnam-clocks"></a>
</h2>
<p>We can compute the correlation among biological clocks using the
function <code>plotCorClocks</code> that requires the package
<code>ggplot2</code> and <code>ggpubr</code> to be installed in your
computer.</p>
<p>We can obtain, for instance, the correlation among the clocks
estimated for the healthy individuals study previosuly analyze (GEO
accession number GSE58045) by simply executing:</p>
<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCorClocks.html">plotCorClocks</a></span><span class="op">(</span><span class="va">age.gse58045</span><span class="op">)</span></span></code></pre></div>
<p><img src="methylclock_files/figure-html/plotCorClockHealth-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<pre class="sessioninfo"><code><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-alfonso2018" class="csl-entry">
Alfonso, Gerardo, and Juan R Gonzalez. 2020. <span>“Bayesian Neural
Networks Improve Methylation Age Estimates.”</span> <em>bioRxiv</em>.
</div>
<div id="ref-bohlin2016prediction" class="csl-entry">
Bohlin, Jon, Siri Eldevik Håberg, Per Magnus, Sarah E Reese, Håkon K
Gjessing, Maria Christine Magnus, Christine Louise Parr, CM Page,
Stephanie J London, and Wenche Nystad. 2016. <span>“Prediction of
Gestational Age Based on Genome-Wide Differentially Methylated
Regions.”</span> <em>Genome Biology</em> 17 (1): 207.
</div>
<div id="ref-chen2017epigenome" class="csl-entry">
Chen, Wei, Ting Wang, Maria Pino-Yanes, Erick Forno, Liming Liang, Qi
Yan, Donglei Hu, et al. 2017. <span>“An Epigenome-Wide Association Study
of Total Serum IgE in Hispanic Children.”</span> <em>Journal of Allergy
and Clinical Immunology</em> 140 (2): 571–77.
</div>
<div id="ref-guintivano2013cell" class="csl-entry">
Guintivano, Jerry, Martin J Aryee, and Zachary A Kaminsky. 2013.
<span>“A Cell Epigenotype Specific Model for the Correction of Brain
Cellular Heterogeneity Bias and Its Application to Age, Brain Region and
Major Depression.”</span> <em>Epigenetics</em> 8 (3): 290–302.
</div>
<div id="ref-haftorn2021epic" class="csl-entry">
Haftorn, Kristine L, Yunsung Lee, William RP Denault, Christian M Page,
Haakon E Nustad, Robert Lyle, Håkon K Gjessing, et al. 2021. <span>“An
EPIC Predictor of Gestational Age and Its Application to Newborns
Conceived by Assisted Reproductive Technologies.”</span> <em>Clinical
Epigenetics</em> 13 (1): 1–13.
</div>
<div id="ref-hannum2013genome" class="csl-entry">
Hannum, Gregory, Justin Guinney, Ling Zhao, Li Zhang, Guy Hughes,
SriniVas Sadda, Brandy Klotzle, et al. 2013. <span>“Genome-Wide
Methylation Profiles Reveal Quantitative Views of Human Aging
Rates.”</span> <em>Molecular Cell</em> 49 (2): 359–67.
</div>
<div id="ref-horvath2013dna" class="csl-entry">
Horvath, Steve. 2013. <span>“DNA Methylation Age of Human Tissues and
Cell Types.”</span> <em>Genome Biology</em> 14 (10): 3156.
</div>
<div id="ref-horvath2018epigenetic" class="csl-entry">
Horvath, Steve, Junko Oshima, George M Martin, Ake T Lu, Austin Quach,
Howard Cohen, Sarah Felton, et al. 2018. <span>“Epigenetic Clock for
Skin and Blood Cells Applied to Hutchinson Gilford Progeria Syndrome and
Ex Vivo Studies.”</span> <em>Aging (Albany NY)</em> 10 (7): 1758.
</div>
<div id="ref-knight2016epigenetic" class="csl-entry">
Knight, Anna K, Jeffrey M Craig, Christiane Theda, Marie Bækvad-Hansen,
Jonas Bybjerg-Grauholm, Christine S Hansen, Mads V Hollegaard, et al.
2016. <span>“An Epigenetic Clock for Gestational Age at Birth Based on
Blood Methylation Data.”</span> <em>Genome Biology</em> 17 (1): 206.
</div>
<div id="ref-lee2019placental" class="csl-entry">
Lee, Yunsung, Sanaa Choufani, Rosanna Weksberg, Samantha L Wilson,
Victor Yuan, Amber Burt, Carmen Marsit, et al. 2019. <span>“Placental
Epigenetic Clocks: Estimating Gestational Age Using Placental DNA
Methylation Levels.”</span> <em>Aging (Albany NY)</em> 11 (12): 4238.
</div>
<div id="ref-levine2018epigenetic" class="csl-entry">
Levine, Morgan E, Ake T Lu, Austin Quach, Brian H Chen, Themistocles L
Assimes, Stefania Bandinelli, Lifang Hou, et al. 2018. <span>“An
Epigenetic Biomarker of Aging for Lifespan and Healthspan.”</span>
<em>Aging (Albany NY)</em> 10 (4): 573.
</div>
<div id="ref-lu2019dna" class="csl-entry">
Lu, Ake T, Anne Seeboth, Pei-Chien Tsai, Dianjianyi Sun, Austin Quach,
Alex P Reiner, Charles Kooperberg, et al. 2019. <span>“DNA
Methylation-Based Estimator of Telomere Length.”</span> <em>Aging
(Albany NY)</em> 11 (16): 5895.
</div>
<div id="ref-mayne2017accelerated" class="csl-entry">
Mayne, Benjamin T, Shalem Y Leemaqz, Alicia K Smith, James Breen, Claire
T Roberts, and Tina Bianco-Miotto. 2017. <span>“Accelerated Placental
Aging in Early Onset Preeclampsia Pregnancies Identified by DNA
Methylation.”</span> <em>Epigenomics</em> 9 (3): 279–89.
</div>
<div id="ref-mcewen2019pedbe" class="csl-entry">
McEwen, Lisa M, Kieran J O?Donnell, Megan G McGill, Rachel D Edgar,
Meaghan J Jones, Julia L MacIsaac, David Tse Shen Lin, et al. 2019.
<span>“The PedBE Clock Accurately Estimates DNA Methylation Age in
Pediatric Buccal Cells.”</span> <em>Proceedings of the National Academy
of Sciences</em>, 201820843.
</div>
<div id="ref-min2018meffil" class="csl-entry">
Min, JL, G Hemani, G Davey Smith, C Relton, M Suderman, and John
Hancock. 2018. <span>“Meffil: Efficient Normalization and Analysis of
Very Large DNA Methylation Datasets.”</span> <em>Bioinformatics</em>.
</div>
<div id="ref-reinius2012differential" class="csl-entry">
Reinius, Lovisa E, Nathalie Acevedo, Maaike Joerink, Göran Pershagen,
Sven-Erik Dahlén, Dario Greco, Cilla Söderhäll, Annika Scheynius, and
Juha Kere. 2012. <span>“Differential DNA Methylation in Purified Human
Blood Cells: Implications for Cell Lineage and Studies on Disease
Susceptibility.”</span> <em>PloS One</em> 7 (7): e41361.
</div>
<div id="ref-teschendorff2012beta" class="csl-entry">
Teschendorff, Andrew E, Francesco Marabita, Matthias Lechner, Thomas
Bartlett, Jesper Tegner, David Gomez-Cabrero, and Stephan Beck. 2012.
<span>“A Beta-Mixture Quantile Normalization Method for Correcting Probe
Design Bias in Illumina Infinium 450 k DNA Methylation Data.”</span>
<em>Bioinformatics</em> 29 (2): 189–96.
</div>
<div id="ref-wang2015systematic" class="csl-entry">
Wang, Ting, Weihua Guan, Jerome Lin, Nadia Boutaoui, Glorisa Canino,
Jianhua Luo, Juan Carlos Celedón, and Wei Chen. 2015. <span>“A
Systematic Study of Normalization Methods for Infinium 450k Methylation
Data Using Whole-Genome Bisulfite Sequencing Data.”</span>
<em>Epigenetics</em> 10 (7): 662–69.
</div>
<div id="ref-wu2019dna" class="csl-entry">
Wu, Xiaohui, Weidan Chen, Fangqin Lin, Qingsheng Huang, Jiayong Zhong,
Huan Gao, Yanyan Song, and Huiying Liang. 2019. <span>“DNA Methylation
Profile Is a Quantitative Measure of Biological Aging in
Children.”</span> <em>Aging (Albany NY)</em> 11 (22): 10031.
</div>
<div id="ref-zhang2019improved" class="csl-entry">
Zhang, Qian, Costanza L Vallerga, Rosie M Walker, Tian Lin, Anjali K
Henders, Grant W Montgomery, Ji He, et al. 2019. <span>“Improved
Precision of Epigenetic Clock Estimates Across Tissues and Its
Implication for Biological Ageing.”</span> <em>Genome Medicine</em> 11
(1): 1–11.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Dolors Pelegri-Siso, Juan R. Gonzalez.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
